### PR TITLE
[Angular] XM Cloud Forms support

### PR DESCRIPTION
**Pull request by illiakovalenko**
- [x] This PR follows the [Contribution Guide](https://github.com/Sitecore/jss/blob/dev/CONTRIBUTING.md)
- [x] Changelog updated
- [x] Upgrade guide entry added

## Description / Motivation
* Introduced a new Form component:
  * The component is a wrapper around the built Sitecore Form that allows you to fetch and render the form from Sitecore Edge.
  * The component is a part of the @sitecore-jss/sitecore-jss-angular package.
  * Component is rendered only on the client side.
  * Handles the case when FormId is not provided, custom error message is displayed.
  * Handles the case when network error occurs during fetching the form from Sitecore Edge, custom error message is displayed.
* Introduced a new InjectionToken EDGE_CONFIG to provide the configuration for the Form component to be able to fetch the form from Sitecore Edge.
* Introduced a plugins technique for the component factory generation to add extensebility for add-ons.
* Introduced a new "getEdgeProxyFormsUrl" method to get the URL for fetching the form from Sitecore Edge.

## Testing Details

- [x] Unit Test Added
- [x] Manual Test/Other (Please elaborate)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/sitecore-jss-angular/src/components/form.component.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      );
+
+      this.hasError = true;
+    }
+  }
+
+  /**
+   * When you set the innerHTML property of an element, the browser does not execute any <script> tags included in the HTML string
+   * This method ensures that any <script> elements within the loaded HTML are executed.
+   * It creates new script elements and appends them to the document head, then removes them to avoid duplication.
```
Comment (yavorsk) [inline packages/sitecore-jss-angular/src/components/form.component.ts:115]: should this comment be updated (the 'appends them to the document head' part)?

**Review by yavorsk — APPROVED** [review]
Great work. 
Please see only one comment.

File: packages/sitecore-jss-angular/src/components/form.component.ts (hunk 1 above)
Comment (illiakovalenko) [reply packages/sitecore-jss-angular/src/components/form.component.ts:115]: correct!

File: packages/sitecore-jss-angular/src/components/form.component.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+} & ComponentRendering;
+
+/**
+ * A component that renders a Sitecore Form.
+ * It fetches the form markup from the Sitecore Edge service and renders it in the component's template.
+ */
+@Component({
+  selector: 'app-form',
+  template: `
+    <ng-container *ngIf="!rendering.params.FormId">
```
[the remark this answers is not in the document]
Comment (illiakovalenko) [reply packages/sitecore-jss-angular/src/components/form.component.ts]: Updated, now we show orange/red boxes only in Editing/Preview mode. Instead in production I log warning message

**Pull request merged**